runtime/lua/vim/lsp/structures/init.lua

```diff
@@ -0,0 +1,36 @@
+--[[
+
+structures.lua
+
+structures is the location to put the handling of any of the structures as defined by
+the language server protocol. Any function defined on a structure must take as the first
+argument the structure itself and then any optional parameters required for determining
+the behavior.
+
+For example, all `Location.*` must be a function of the form:
+
+  function(location, ...)
+
+Additionally, within each `structure.*`, they can import only directly from other structures using:
+
+  require('vim.lsp.structures.location')
+
+Rather than:
+
+  require('vim.lsp.structures').location
+
+
+However, from the rest of the project, they should be imported in the latter style.
+
+--]]
+
+local structures = {}
```

Comment:
This is probably the place to add a function `is_type` or similar, for standardized checking of types.